tools/analyzer_plugin/test/integration/test_bases/analysis_driver_test_base.dart

```diff
@@ -0,0 +1,115 @@
+import 'dart:io';
+
+import 'package:analyzer/file_system/file_system.dart';
+import 'package:analyzer/file_system/memory_file_system.dart';
+import 'package:analyzer/file_system/physical_file_system.dart';
+import 'package:analyzer/src/context/context_root.dart';
+import 'package:analyzer/src/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/src/dart/analysis/byte_store.dart';
+import 'package:analyzer/src/dart/analysis/driver.dart'
+    show AnalysisDriver, AnalysisDriverScheduler;
+import 'package:analyzer/src/dart/analysis/driver.dart';
+import 'package:analyzer/src/dart/analysis/file_state.dart';
+import 'package:analyzer/src/dart/analysis/performance_logger.dart';
+import 'package:analyzer/src/generated/engine.dart';
+import 'package:analyzer/src/generated/source.dart';
+import 'package:analyzer/src/generated/source_io.dart';
+import 'package:analyzer/src/source/package_map_resolver.dart';
+import 'package:analyzer/src/test_utilities/mock_sdk.dart';
+import 'package:meta/meta.dart';
+import 'package:package_config/package_config.dart';
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+abstract class AnalysisDriverTestBase {
+  static const _realPackagesToCopyIntoResourceProvider = [
+    'over_react',
+  ];
+
+  AnalysisDriver get analysisDriver => _analysisDriver;
+  AnalysisDriver _analysisDriver;
+
+  MemoryResourceProvider get resourceProvider => _resourceProvider;
+  MemoryResourceProvider _resourceProvider;
+
+  String get testPath => _testPath;
+  String _testPath;
+
+  Source newSource(String path, [String content]) {
+    expect(p.isAbsolute(path), isFalse,
+        reason: 'newSource() must be called with a relative path');
+    final absolutePath = p.join(testPath, path);
+    final file = resourceProvider.newFile(absolutePath, content ?? '');
+    return file.createSource();
+  }
+
+  String sourcePath(Source source) =>
+      resourceProvider.convertPath(source.uri.path);
+
+  @mustCallSuper
+  Future<void> setUp() async {
+    _resourceProvider = MemoryResourceProvider();
+
+    final sdk = MockSdk(resourceProvider: resourceProvider);
+    final packageMap = {
+      for (final packageName in _realPackagesToCopyIntoResourceProvider)
+        packageName: [await _loadRealPackage(packageName, resourceProvider)],
+    };
+
+    // Setup a testing directory. All calls to [newSource] will create files
+    // within this directory.
+    _testPath = resourceProvider.newFolder('/test').path;
+
+    final logger = PerformanceLog(StringBuffer());
+    final analysisScheduler = AnalysisDriverScheduler(logger)..start();
+    _analysisDriver = AnalysisDriver(
```

Comment:
> I noticed that we also create a driver in plugin.dart; could we reuse any of that code here?

_From @greglittlefield-wf on the previous PR_

Reply:
Not a big deal if not